src/messages/TextWithButtons/TextWithButtons.tsx

```diff
@@ -21,6 +21,16 @@ const TextWithButtons: FC = props => {
 	const { action, message, config, onEmitAnalytics, messageParams, openXAppOverlay } =
 		useMessageContext();
 
+	const progressiveMessageRendering = !!config?.settings?.behavior?.progressiveMessageRendering;
+	const botOutputMaxWidthPercentage = config?.settings?.layout?.botOutputMaxWidthPercentage;
+
+	const isBotMessage = message.source === "bot";
+	const isEngagementMessage = message.source === "engagement";
+
+	const [showActionButtons, setShowActionButtons] = useState(
+		progressiveMessageRendering ? false : true,
+	);
```

Comment:
The action buttons are not visible in the previous conversation chat log when progressive rendering is enabled. I doubt that it could be because of this state.

Reply:
good catch! it is totally because of the state. removed local state and added check on animationState instead